product-service/products/get-product-by-id.ts

```diff
@@ -1,19 +1,32 @@
 import * as AWSLambda from "aws-lambda";
 import { CORS_HEADERS } from "../cors-headers";
-import productList from "./products.mock.json";
+import { Client, ClientConfig, QueryResult } from "pg";
 
-interface Product {
-  id: string;
-  title: string;
-  description: string;
-  price: number;
-  count: number;
-}
+const { DB_HOST, DB_PORT, DB_DATABASE, DB_USERNAME, DB_PASSWORD } = process.env;
+const dbOptions: ClientConfig = {
+  host: DB_HOST,
+  port: +DB_PORT,
+  database: DB_DATABASE,
+  user: DB_USERNAME,
+  password: DB_PASSWORD,
+  ssl: { rejectUnauthorized: false },
+  connectionTimeoutMillis: 5000,
+};
 
 export const getProductByID = async (event: AWSLambda.APIGatewayEvent) => {
+  const client = new Client(dbOptions);
+
   try {
-    const product: Product = productList.products.find((product) => product.id === event.pathParameters.productId);
-    if (!product) {
+    await client.connect();
+    const product_id = event.pathParameters.productId;
+    const {
+      rows,
+    }: QueryResult<Product> = await client.query(
+      "select count, title, description, price, product_id id from products p left join stocks s on $1 = s.product_id where p.id = $1",
+      [product_id]
+    );
+    console.log(`SUCCESS get product ${product_id}: `, rows);
+    if (!rows) {
       return {
         statusCode: 500,
```

Comment:
Not related to the current task, but anyway, I tend to feel that HTTP status 404 would fit better here rather than 500